### ISSUE TITLE
F-string still aren't working correctly.

### ISSUE THREAD

**Issue by MrQubo** [xonsh/xonsh#2870]
The #2829 was a fix for #2822. But it handles only a simple case where the variable is the only thing inside braces which is a small subset of possible usage. The code inside braces might be almost any valid xonsh expression.
The #2829 also breaks some things. For example, the first two cases below which is escaping braces with double braces and using f-strings inside an f-string.

Here I provide some cases that aren't currently handled correctly:
```
f"{{$X}}"
f"{f'{$X}'}"
f"{$X + $Y}"
f"{${'X'}}"
f"{repr($X)}"
f"{$(uname)}"
```

Those cases _are_ currently handled correctly but might be useful for testing:
```
f"{'$X'}"
f"a{$X}b"
f"{'}'}"
f"{'$'}"
```

As mentioned in #2829 we might need to parse the f-string content to correctly handle all possible cases. As a reference for implementation, we could check the f-string related code from the cpython's [parser implementation](https://github.com/python/cpython/blob/656d52dbfde3223cd2a3525d652b6cccb02fa991/Python/ast.c#L4927).

**Comment by scopatz** [comment]
Yep, you are right that this needs to be more fully implemented

**Comment by scopatz** [comment]
Thanks for reporting @MrQubo!

**Comment by scopatz** [comment]
Potential fix is available in #2892
